app/models/talk.rb

```diff
@@ -103,6 +103,8 @@ def happened?
   private
 
   def send_slack_notification
+    return if ENV['SLACK_WEBHOOK_URL'].blank?
+
```

Comment:
Rather than preventing from sending the message what about sending it to a test channel ?

To be tested but I think something like that would work
```ruby
body = {
  channel: Rails.env.production? ? "orga" : "orga_test",
  text: "Nouveau talk: #{title} par #{speaker_name}"
}

conn.post(ENV['SLACK_WEBHOOK_URL'], body.to_json)
```

Reply:
Not sure we want every review app to send notification, we can still allow notification for one review app is necessary.

Reply:
How can we then test the form from a preview app ?
I think at some point this is a part we will want to redesign so being able to test it would be great.

Reply:
By modifying `SLACK_WEBHOOK_URL` value of the app

Reply:
ok looks like I didn't realised that notification are sent in an after_save callback. So seeds do generate notifications that is not so good indeed... Could we try to bypass the after_save during seed or remove the callback from the model and make an explicit call in the controller.

## testing webhook
So if we want to test a the form submission for a PR we need to : 
- create a new slack URL that will post to another channel as we probably don't want top spam `#orga` with tests
- modify  the env value in scalingo.json when we test
- change back the env value before merging that branch
I am assuming that who ever opens the PR does not have access to Scalingo admin
or ping you to set up the url

do I get it right ?

Reply:
> So if we want to test a the form submission for a PR we need to :

1. Change the channel at then end of the slack URL (the URL looks like this `https://slack.com/blablabla#orga`)
2. Set the env variable directly in the app on Scalingo, no need to update the scalingo.json file

> Could we try to bypass the after_save during seed or remove the callback from the model and make an explicit call in the controller.

Gonna do that indeed